ecdsa/resharing/round_3_old_step_2.go

```diff
@@ -33,8 +32,12 @@ func (round *round3) Start() *tss.Error {
 	for j, Pj := range round.NewParties().IDs() {
 		share := round.temp.NewShares[j]
 		r3msg1 := NewDGRound3Message1(Pj, round.PartyID(), share)
-		round.temp.dgRound3Message1s[i] = r3msg1
```

Comment:
In the line below library sends FeldmanVSS shares generated in [round 1](https://github.com/timoth-y/tss-lib/blob/ecdsa/reshare_issues/ecdsa/resharing/round_1_old_step_1.go#L56) directly to each player Pj in the new committee, so that they could verify their consistency with their own ones [here](https://github.com/timoth-y/tss-lib/blob/ecdsa/reshare_issues/ecdsa/resharing/round_4_new_step_2.go#L139-L142) in round 4 (they use commitments Cj received during round 1 and decommitments Dj send to each other in the lines below).

The problem here is that the library caches messages for each other player Pj as if one of these messages (last one) is directed to itself (by index `i`). As a result, all but players with the highest index would have a faulty cache `round.temp.dgRound3Message1s[i]`.

This would backfire in the following round during that share consistency checks [here](https://github.com/timoth-y/tss-lib/blob/ecdsa/reshare_issues/ecdsa/resharing/round_4_new_step_2.go#L139-L142). This if\else condition should do the trick right.